File: GO/Cocodrilo.go
package main

import "fmt"

type Cocodrilo struct {
	BaseReptiles
}

// Constructor con parametros
func NewCocodrilo(nombre, piel string) * Cocodrilo {
	return &Cocodrilo{
		BaseReptiles: *NewBaseReptiles(nombre, piel),
	}
}

// Constructor sin parametros
func NewCocodriloSinParametros() *Cocodrilo {
	return &Cocodrilo{
		BaseReptiles: *NewBaseReptiles("Cocodrilo Dante", "Piel seca y escamosa"),
	}
}

// Constructor copia
func NewCocodriloCopia(copia *Cocodrilo) *Cocodrilo {
	return &Cocodrilo{
		BaseReptiles: *NewBaseReptiles(
			copia.Nombre,
			copia.Piel,
		),
	}
}

// Destructor
func (c *Cocodrilo) Close() error {
	fmt.Printf("Destructor...")
	return nil
}

// Metodo de la clase padre
func (c *Cocodrilo) PonerHuevos() string {
	return "Ponen entre 14 y 40 huevos al año."
}

// Metodos de las interfaces
func (c *Cocodrilo) Habitat() string {
	return "Rios."
}

func (c *Cocodrilo) Reproducirse() string {
	return "Se reproducen sexualmente, son oviparos."
}

File: JAVASCRIPT/clases/Cocodrilo.js
import { Reptiles } from "../clases-abstractas/Reptiles.js"
import { IReproducirse } from "../interfaces/IReproducirse.js"
import { IHabitat } from "../interfaces/IHabitat.js"

export class Cocodrilo extends Reptiles {
    // Constructor con parametros
    constructor(nombre = "Cocodrilo Dante", piel  ="Piel seca y escamosa") {
        super(nombre, piel)
    }

    // Constructor copia
    clone() {
        return new Cocodrilo(this.nombre, this.piel)
    }

    // Destructor
    destroy() {
        console.log("Destructor...")
        this.nombre = false
        this.piel = false
    }

    // Metodo de la clase padre
    ponerHuevos() {
        console.log("Ponen entre 14 y 40 huevos al año.")
    }

    // Metodos implementados de las interfaces
    reproducirse() {
        console.log("Se reproducen sexualmente, son oviparos.")
    }

    habitat() {
        console.log("Rios.")
    }
}